### PR TITLE
FIX-#6948: Fix groupby when Modin dataframe has several column partitions

### LINKED ISSUE #6948: Poor Performance on TPC-H Queries
In a benchmark project ([link](https://github.com/hesamshahrokhi/modin_tpch_benchmark)), I tried to compare the performance of Pandas and Modin (on Ray and Dask) on simple TPC-H queries (Q1 and Q6) written in Pandas. The results ([link](https://github.com/hesamshahrokhi/modin_tpch_benchmark/blob/main/Results_TPCH_SF1.txt)) show that the ```read_table``` API is 2x faster than its Pandas equivalent on Ray. However, the same API on Dask, and also the individual query run times on both backends are much slower than Pandas. Moreover, the results of Q1 are not correct.
 
Could you please help me to understand why this is the case? I tried to use the suggested configurations and imported the data using ```read_table``` which is officially supported by Modin.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6948 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/groupby.py [hunk 1]
```diff
@@ -904,19 +905,37 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
...
+                            labels[idx] = x if not callable(x) else x.__name__
+                            continue
+                        new_elem = []
+                        for y in x:
+                            if is_scalar(y) or callable(y):
+                                new_elem.append(y if not callable(y) else y.__name__)
+                            else:
+                                new_elem.extend(y)
+                        for el in new_elem:
+                            assert is_scalar(el)
```
Comment (anmyachev) [inline modin/pandas/groupby.py:931]:
```suggestion
```

File: modin/pandas/groupby.py [hunk 2]
```diff
@@ -904,19 +905,35 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
...
                 def do_relabel(obj_to_relabel):  # noqa: F811
-                    new_order, new_columns_idx = order, pandas.Index(new_columns)
+                    # unwrap nested labels into one level tuple
+                    labels = [None] * len(old_kwargs)
+                    for idx, x in enumerate(old_kwargs.values()):
+                        if is_scalar(x) or callable(x):
+                            labels[idx] = x if not callable(x) else x.__name__
+                            continue
+                        new_elem = []
+                        for y in x:
```
Comment (YarShev) [inline modin/pandas/groupby.py:925]: Let's give x and y more descriptive names.

Comment (anmyachev) [reply modin/pandas/groupby.py]: renamed

File: modin/pandas/groupby.py [hunk 3]
```diff
@@ -904,19 +905,35 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
 
         do_relabel = None
         if isinstance(func, dict) or func is None:
-            relabeling_required, func_dict, new_columns, order = reconstruct_func(
+            # the order from `reconstruct_func` cannot be used correctly if there
+            # is more than one columnar partition
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/pandas/groupby.py]: done

**Pull request merged**